examples/MarksDynamicUpdate.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState } from 'react';
+import { Range, getTrackBackground } from '../src/index';
+
+const STEP = 1;
+const MIN = 0;
+
+const UpdatingMarks = () => {
+  const [values, setValues] = React.useState([50])
+  const [selectedMax, setSelectedMax] = useState((100))
+  const [maxOptions] = useState([100, 150, 200, 250, 300]) 
+
+  const [selectedMin, setSelectedMin] = useState((0))
+  const [minOptions] = useState([0, 15, 20, 25, 30]) 
+
+
+      return (
+      <div
+        style={{
+          display: 'flex',
+          justifyContent: 'center',
+          flexWrap: 'wrap'
+        }}
+      >
+        <Range
+          values={values}
+          setValues={setValues}
```

Comment:
Why this new API and not invoking `onChange` instead?

Reply:
setValues is an optional prop (callback). This was my take on solving for when I change the range max to, say, 300, select a value of [280], then change range max back to something smaller than 280 like a max of 100. In componentDidUpdate, I use a helper function to determine if that value falls outside my range max and min, and if it does, I replace it with either the max or the min, depending on where the excedence falls and if the user passed a callback for setValues, I invoke it with the new values so my range slider never represents a selectedValue that isn't currently present on the dynamically shifted range boundaries. 

So onChange would only occur if the user explicitly makes a selection. This would only be invoked in the event that the users selection no longer represents the current state of the range.

Reply:
I suppose I could invoke onChange, but my original values array would remain the same/out of date.

Reply:
Oh, ok. I see now. Yes - makes absolute sense to only invoke onChange.